actors/user/src/main/java/org/sunbird/user/actors/UserProfileActor.java

```diff
@@ -0,0 +1,265 @@
+
+    Map<String, Object> esResult = getUserByIdFromES(userId);
+    Map<String, Object> esProfileVisibility = getProfileVisibilityByUserIdFromES(userId);
+
+    esProfileVisibility =
+        handlePublicToPrivateConversion(privateList, esResult, esProfileVisibility);
+
+    handlePrivateToPublicConversion(publicList, esResult, esProfileVisibility);
+    updaeProfileVisibility(userId, privateList, publicList, esResult);
+
+    updateDataInES(esResult, esProfileVisibility, userId);
+    Response response = new Response();
+    response.put(JsonKey.RESPONSE, JsonKey.SUCCESS);
+    sender().tell(response, self());
+    generateTeleEventForUser(null, userId, "profileVisibility");
+  }
+
+  private void updaeProfileVisibility(
+      String userId,
+      List<String> privateList,
+      List<String> publicList,
+      Map<String, Object> esResult) {
+    Map<String, String> profileVisibilityMap =
+        (Map<String, String>) esResult.get(JsonKey.PROFILE_VISIBILITY);
+    if (null == profileVisibilityMap) {
+      profileVisibilityMap = new HashMap<>();
+    }
+    prepareProfileVisibilityMap(profileVisibilityMap, privateList, JsonKey.PRIVATE);
+    prepareProfileVisibilityMap(profileVisibilityMap, publicList, JsonKey.PUBLIC);
+
+    if (profileVisibilityMap.size() > 0) {
+      updateCassandraWithPrivateFiled(userId, profileVisibilityMap);
+      esResult.put(JsonKey.PROFILE_VISIBILITY, profileVisibilityMap);
+    }
+  }
+
+  private void prepareProfileVisibilityMap(
+      Map<String, String> profileVisibilityMap, List<String> list, String value) {
+    if (list != null) {
+      for (String key : list) {
+        profileVisibilityMap.put(key, value);
+      }
+    }
+  }
+
+  private void handlePrivateToPublicConversion(
+      List<String> publicList,
+      Map<String, Object> esResult,
+      Map<String, Object> esProfileVisibility) { // now have a check for public field.
+    if (publicList != null && !publicList.isEmpty()) {
+      // this estype will hold all private data of user.
+      // now collecting values from private filed and it will update
+      // under original index with public field.
+      for (String field : publicList) {
+        if (esProfileVisibility.containsKey(field)) {
+          esResult.put(field, esProfileVisibility.get(field));
+          esProfileVisibility.remove(field);
+        } else {
+          ProjectLogger.log("field value not found inside private index ==" + field);
+        }
+      }
+    }
+  }
+
+  private Map<String, Object> handlePublicToPrivateConversion(
+      List<String> privateList,
+      Map<String, Object> esResult,
+      Map<String, Object> esProfileVisibility) {
+    Map<String, Object> privateDataMap = null;
+    if (privateList != null && !privateList.isEmpty()) {
+      privateDataMap = handlePrivateVisibility(privateList, esResult, esProfileVisibility);
+    }
+    if (privateDataMap != null && privateDataMap.size() >= esProfileVisibility.size()) {
```

Comment:
Is size comparison correct? What if the count is less but different fields are mentioned?